drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/AbstractScenarioSimulationDocksHandler.java

```diff
@@ -0,0 +1,164 @@
+        cheatSheetDock = new UberfireDock(UberfireDockPosition.EAST,
+                                          "FILE_TEXT",
+                                          new DefaultPlaceRequest(CheatSheetPresenter.IDENTIFIER),
+                                          perspectiveIdentifier);
+        result.add(cheatSheetDock.withSize(450).withLabel(ScenarioSimulationEditorConstants.INSTANCE.scenarioCheatSheet()));
+        return result;
+    }
+
+    public void addDocks() {
+        refreshDocks(true, false);
+    }
+
+    public void removeDocks() {
+        refreshDocks(true, true);
+    }
+
+    public void expandToolsDock() {
+        authoringWorkbenchDocks.expandAuthoringDock(toolsDock);
+    }
+
+    public abstract void expandTestResultsDock();
+
+    public void resetDocks() {
+        getSettingsPresenter().ifPresent(SubDockView.Presenter::reset);
+        getCheatSheetPresenter().ifPresent(SubDockView.Presenter::reset);
+        getTestToolsPresenter().ifPresent(SubDockView.Presenter::reset);
+    }
+
+    public void setScesimEditorId(String scesimEditorId) {
+        currentScesimEditorId = scesimEditorId;
+        settingsDock.getPlaceRequest().addParameter(SCESIMEDITOR_ID, scesimEditorId);
+        toolsDock.getPlaceRequest().addParameter(SCESIMEDITOR_ID, scesimEditorId);
+        cheatSheetDock.getPlaceRequest().addParameter(SCESIMEDITOR_ID, scesimEditorId);
+    }
+
+    public Optional<CheatSheetView.Presenter> getCheatSheetPresenter() {
+        final Optional<CheatSheetView> cheatSheetView = getCheatSheetView(getCurrentRightDockPlaceRequest(CheatSheetPresenter.IDENTIFIER));
+        return cheatSheetView.map(CheatSheetView::getPresenter);
+    }
+
+    public Optional<TestToolsView.Presenter> getTestToolsPresenter() {
+        final Optional<TestToolsView> testToolsView = getTestToolsView(getCurrentRightDockPlaceRequest(TestToolsPresenter.IDENTIFIER));
+        return testToolsView.map(TestToolsView::getPresenter);
+    }
+
+    public Optional<SettingsView.Presenter> getSettingsPresenter() {
+        final Optional<SettingsView> settingsView = getSettingsView(getCurrentRightDockPlaceRequest(SettingsPresenter.IDENTIFIER));
+        return settingsView.map(SettingsView::getPresenter);
+    }
+
+    /**
+     * Returns a <code>PlaceRequest</code> for the <b>status</b> of the right dock with the given <b>identifier</b>
+     * relative to the current instance of <code>ScenarioSimulationEditorPresenter</code>
+     * @return
+     */
+    protected PlaceRequest getCurrentRightDockPlaceRequest(String identifier) {
+        PlaceRequest toReturn = new DefaultPlaceRequest(identifier);
+        toReturn.addParameter(SCESIMEDITOR_ID, String.valueOf(currentScesimEditorId));
+        return toReturn;
+    }
+
+    protected Optional<TestToolsView> getTestToolsView(PlaceRequest placeRequest) {
+        final Activity activity = placeManager.getActivity(placeRequest);
+        if (activity != null) {
```

Comment:
Indirect condition.
Could be replaced 
if (activity == null) {
 return Optional.empty();}
else{
final AbstractWorkbenchActivity testToolsActivity = (AbstractWorkbenchActivity) activity;
            return Optional.of((TestToolsView) testToolsActivity.getWidget());
}